### ISSUE TITLE
Bug: AI destroys & rebuilds stables if not able to reach them

### ISSUE THREAD

**Issue by Halbarad1** [Sh0wdown/UnofficialCrusaderPatch#652]
Hi,
If you got stables in your AIV and the AI isn't able to reach them, they destroy them. After that they do rebuild them. One original Castle of the Marshal had that problem, crippling him for the whole game, since he was spending most of his money for that stupid stable that he immediately destroyed getting half the price back. 
Got a small reminder when a similar thing happend to an AI created by myself, after the way to the stables was seen as "blocked" because the AI build a hop farm on the way to the stable (funny thing: The stables directly in front of the farm weren't affected, only the ones whose way was "blocked" by the hop farm).
Solution: AI should see Stables the same way as houses: A way shouldn't be required. 
I'm not quite sure if I postet this bug, but since I didn't found it in Bugs probably not. 

Deutsch: Wenn Ställe einer KI-Burg (AIV) nicht zugänglich sind, baut und reißt die KI die Ställe ununterbrochen ab. Das kostet sie meist ihr gesamtes Geld. Eine orginale Marshallburg hat das Problem. Ist mir wieder aufgefallen, als eine eigens erstellte KI den Weg zum Stall mit einer Hopfenfarm blockiert hat und ihn deshalb ununterbrochen auf- und abgebaut hat. Die Ställe direkt vor der Hopfenfarm waren nicht betroffen.  

The picture shows where the stables should be (blue mark, right one even shows the clouds because one of them just got demolished) and where the hop farm stand in their way (red mark). The other stables are fine as you can see. 
![Screenshot (6)_LI](https://user-images.githubusercontent.com/44904062/94010145-d39e0080-fda5-11ea-96be-d0bcdb29523a.jpg)

**Comment by Krarilotus** [comment]
@Halbarad1 did you enable the 'Don't destroy unaccessable buildings' option in the UCP yet?

**Comment by Halbarad1** [comment]
Yes I did. Although I'm not quite sure if it really does work, since my AI Also destroyed other buildings (cathedral e. g.) when there wasn't a way to them. 
Does it work for you?

**Comment by Krarilotus** [comment]
FOr me this usually worked fine. There are two options tho for this:
One is hidden in the AI tab, where you have to manually tick the boxes for enclosed buildings not to be destroyed.

I am not quite sure if both the options in the patcher overlap

**Comment by Halbarad1** [comment]
Well there is 1. the bugfix that AI doesnt destroy buildings it can not reach and 2. the AI option, that it doesn't destroy buildings when being in danger. I activated both options. I made my tick on both of them.
Would be interesting if others don't have this problem, should be easy to check (build an AIV-Castle without gate for the AI with stables and cathedral build outside the walls)

**Comment by Truetobi91** [comment]
I have disabled any type of self destruction in my setup and I don't think I ever encountered an issue again. However I realized I had to make them accessable for a different reason: Fire. If an inaccessable building catches fire it can spread that fire for as long as it burns as the firefighters can't extinguish the flames. Especially in a case like yours where the stables is placed right next to other flamable buildings.

**Comment by Halbarad1** [comment]
I am pretty sure that everyone got this problem since I did check out my settings and my game shouldn't be different to everybody elses game. Check out this AIV in deathmatch mode for any AI: https://drive.google.com/file/d/12rjVvbNooS1l95sztnQP1FP6TxeF5vA2/view?usp=sharing
The AI will build up and immediately destroy its cathedral and stables until gold runs out, the marked will still stay there. 

@Truetobi91 Fire doesn't matter. 
Besides the fact that castles don't always have to be fireproof/ flawless, in the picture above you can see that the stables in the AIV were reachable, only because of the (random) farm placement the AI thought they were not and therefore destroyed them. This specific case just happens in about one of fifty times but if it does it really cripples the AI because it constantly looses 250 gold + wood. Building the stables at a place, where the fire wont spread to other buildings, while the stables themself are surrounded by moat or walls, isn't an option either because of the bug.

**Comment by Heroesflorian** [comment]
@Halbarad1 a couple remarks on this issue: 

> since he was spending most of his money for that stupid stable that he immediately destroyed getting half the price back

actually, iirc, the AI gets back nothing when destroying inaccessible buildings (I know, makes it even worse, yay...)

> funny thing: The stables directly in front of the farm weren't affected, only the ones whose way was "blocked" by the hop farm

Actually 100% explainable: Hops tiles are (unlike wheat tiles) not walkable, and some hops tiles are on the edge of the overall hops farm area, and in this case they line up with the moat to make that one stable perfectly inaccessible (if you tried the same as a human player, you would not be able to send troops there either). So while the other stables were still reachable, the one "behind" it was not. 

Also, as others already pointed out, if all the settings (including relevant sub-options!) related to disabling demolishing of (inaccessible) buildings are ticked, this should no longer occur. I also tested AIVs with fully walled-in oil smelters and workshops a while ago, they did remain (i.e. were not deleted) with those options ticked, for me. 

Having said all that, though, I 100% agree: 

> AI should see Stables the same way as houses: A way shouldn't be required.

**Comment by Halbarad1** [comment]
> Also, as others already pointed out, if all the settings (including relevant sub-options!) related to disabling demolishing of (inaccessible) buildings are ticked, this should no longer occur. 

As I said I made the tick in any setting related to the problem, including the sub-options. If you think that this can't be possible check out my linked AIV, its not the one from the picture, I just made that one so you can see that the problem does exist with every option activated. 
https://drive.google.com/file/d/12rjVvbNooS1l95sztnQP1FP6TxeF5vA2/view?usp=sharing
